### PR TITLE
Cleanup metrics in infrastructure

### PR DESCRIPTION
**Pull request by no2chem**
## Overview

Description: This PR addresses technical debt related to metrics collection in the Corfu infrastructure. This PR removes the isMetricsEnabled parameter on each server handler, as well as the unneeded opTimer class. Instead, in CorfuMsgHandler, metrics status is checked using the static method ```MetricsUtils.isMetricsCollectionEnabled'''. The former opTimer field automatically uses the name of the message from the CorfuMsgType enum. Many handlers had the incorrect opTimer name - probably due to copy-paster error.

Why should this be merged: Address technical debt, reduce complexity around metrics collection.

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/CorfuMsgHandler.java [hunk 1]
```diff
@@ -75,95 +98,151 @@ public CorfuMsgHandler() {
...
+     * @return              A new {@link Handler} which conditionally executes the handler
+     *                      based on preconditions (whether the server is shutdown/ready).
+     */
+    private Handler<CorfuMsg> generateConditionalHandler(@Nonnull final AbstractServer server,
+            @Nonnull final CorfuMsgType type,
+            @Nonnull final Handler<CorfuMsg> handler) {
+        if (!MetricsUtils.isMetricsCollectionEnabled()) {
+            // If metrics are disabled, register the handler without instrumentation.
+            return (msg, ctx, r) -> {
+                if (server.isShutdown()) {
```
Comment (medhavidhawan) [inline infrastructure/src/main/java/org/corfudb/infrastructure/CorfuMsgHandler.java]: I think this logic should go somewhere more common and visible. It should be used in generation but if possible can we keep it in a separate method of it's own.

**Comment by medhavidhawan** [comment]
This probably will change metrics names a little bit but I hope it is ok.

**Comment by no2chem** [comment]
@medhavidhawan No. I'm not sure if this is something we want to test yet, but we should address it when we decide on what to do with metrics. This is more just code cleanup here.

File: infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java [hunk 2]
```diff
@@ -126,8 +126,8 @@
      * Handler for this server.
      */
     @Getter
-    private CorfuMsgHandler handler = new CorfuMsgHandler()
-            .generateHandlers(MethodHandles.lookup(), this);
+    private final CorfuMsgHandler handler =
+        CorfuMsgHandler.generateHandler(MethodHandles.lookup(), this);
 
     private static final String metricsPrefix = "corfu.server.sequencer.";
     private static Counter counterTokenSum;
```
Comment (corfu-repo) [inline infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: MAJOR Remove this unused "counterTokenSum" private field. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)

**Comment by corfu-repo** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ManagementServer.java#L72](https://github.com/CorfuDB/CorfuDB/blob/767b787f7ab0cf865cfa1b07147d9416e73763fa/infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java#L72): Remove this unused "metricsPrefix" private field. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [SequencerServer.java#L134](https://github.com/CorfuDB/CorfuDB/blob/767b787f7ab0cf865cfa1b07147d9416e73763fa/infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java#L134): Remove this unused "counterToken0" private field. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)
1. ![MINOR][MINOR] [AbstractServer.java#L8](https://github.com/CorfuDB/CorfuDB/blob/767b787f7ab0cf865cfa1b07147d9416e73763fa/infrastructure/src/main/java/org/corfudb/infrastructure/AbstractServer.java#L8): Remove this unused import 'org.corfudb.protocols.wireprotocol.CorfuMsgType'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [AbstractServer.java#L9](https://github.com/CorfuDB/CorfuDB/blob/767b787f7ab0cf865cfa1b07147d9416e73763fa/infrastructure/src/main/java/org/corfudb/infrastructure/AbstractServer.java#L9): Remove this unused import 'org.corfudb.util.MetricsUtils'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [SequencerServer.java#L36](https://github.com/CorfuDB/CorfuDB/blob/767b787f7ab0cf865cfa1b07147d9416e73763fa/infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java#L36): Remove this unused import 'org.corfudb.util.MetricsUtils'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**